src/Instances/New.lua

```diff
@@ -7,368 +7,29 @@
-						if ref.instance == nil then
-							if ENABLE_EXPERIMENTAL_GC_MODE then
-								if conn.Connected then
-									warn("ref is nil and instance is around!!!")
-								else
-									print("ref is nil, but instance was destroyed")
-								end
-							end
-							return
-						end
-						value(ref.instance[key.key])
-					end
-
-				-- Unknown symbol key
-				else
-					logError("unrecognisedPropertyKey", nil, key.name)
-				end
-
-			-- Unknown key of arbitrary type
-			else
-				logError("unrecognisedPropertyKey", nil, typeof(key))
+		local classDefaults = defaultProps[className]
+		if classDefaults ~= nil then
+			for defaultProp, defaultValue in pairs(classDefaults) do
+				instance[defaultProp] = defaultValue
 			end
 		end
 
-		--[[
-			STEP 3: If provided, parent [Children] to instance
-		]]
-		local children = propertyTable[Children]
-		if children ~= nil then
-			local currentChildren = {}
-			local prevChildren = {}
-
-			local currentConnections = {}
-			local prevConnections = {}
-
-			local function updateCurrentlyParented()
-				if ref.instance == nil then
-					if ENABLE_EXPERIMENTAL_GC_MODE then
-						if conn.Connected then
-							warn("ref is nil and instance is around!!!")
-						else
-							print("ref is nil, but instance was destroyed")
-						end
-					end
-					return
-				end
-
-				prevChildren, currentChildren = currentChildren, prevChildren
-				prevConnections, currentConnections = currentConnections, prevConnections
-
-				local function recursiveAddChild(child)
-					local childType = typeof(child)
-
-					if childType == "Instance" then
-						-- single instance child
-
-						currentChildren[child] = true
-
-						-- reused or newly parented logic
-						if prevChildren[child] == nil then
-							if overrideParents[child] == nil then
-								child.Parent = ref.instance
-							end
-						else
-							prevChildren[child] = nil
-						end
-
-					elseif childType == "table" then
-						-- could either be an array or state object
-
-						if child.type == "State" then
-							-- state object
-
-							recursiveAddChild(child:get(false))
-
-							-- reuse old connection change handler if possible
-							local prevDisconnect = prevConnections[child]
-							if prevDisconnect ~= nil then
-								currentConnections[child] = prevDisconnect
-								prevConnections[child] = nil
-							else
-								-- FUTURE: does this need to be cleaned up when
-								-- the instance is destroyed at any point?
-								-- If so, how?
-								currentConnections[child] = Observer(child):onChange(function()
-									Scheduler.enqueueCallback(updateCurrentlyParented)
-								end)
-							end
-						else
-							-- array of children
-							for _, subChild in pairs(child) do
-								recursiveAddChild(subChild)
-							end
-						end
-
-					-- explicitly allow nils (probably inside a state object)
-					elseif childType ~= "nil" then
-						logWarn("unrecognisedChildType", childType)
-					end
```

Comment:
@Elttob This is a regression regarding Children handling - nil now throws a warning with the specialKey version

Reply:
Thanks - will push a fix for this asap :)

Reply:
This is interesting; it should already have been accounted for here (in Children.lua):

```Lua
elseif kind == "State" then
-- case 2; state object

local value = child:get(false)
-- allow nil to represent the absence of a child
if value ~= nil then
	processChild(value, autoName)
end
```

Will investigate.

Reply:
Under what circumstance will this break user-facing code? The only time `processChild` should ever have gotten called with `nil` _was_ the State case.

Reply:
I did an oopsie and forgot that Lua won't iterate over keys that have nil. I think all nil cases are handled then.